tests/nonbonded/test_atom_by_atom_energies.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+import pytest
+from common import assert_energy_arrays_match
+
+from timemachine.constants import DEFAULT_TEMP
+from timemachine.ff import Forcefield
+from timemachine.ff.handlers import openmm_deserializer
+from timemachine.lib import custom_ops
+from timemachine.md import builders
+from timemachine.md.barostat.utils import get_bond_list, get_group_indices
+from timemachine.md.exchange.exchange_mover import BDExchangeMove
+from timemachine.potentials import HarmonicBond, Nonbonded
+
+
+@pytest.mark.memcheck
+@pytest.mark.parametrize(
+    "precision,atol,rtol,threshold", [(np.float64, 1e-8, 1e-8, 1e16), (np.float32, 1e-4, 2e-3, 1e8)]
+)
+def test_nonbonded_atom_by_atom_energies_match(precision, atol, rtol, threshold):
+    """Verify that if looking at not computing the subsets of energies matches the references"""
+    rng = np.random.default_rng(2023)
+    ff = Forcefield.load_default()
+    system, conf, box, _ = builders.build_water_system(4.0, ff.water_ff)
+    bps, _ = openmm_deserializer.deserialize_system(system, cutoff=1.2)
+    nb = next(bp for bp in bps if isinstance(bp.potential, Nonbonded))
+    bond_pot = next(bp for bp in bps if isinstance(bp.potential, HarmonicBond)).potential
+
+    all_group_idxs = get_group_indices(get_bond_list(bond_pot), conf.shape[0])
+
+    # Get the indices of atoms to compute per atom energies for
+    target_atoms = rng.choice(all_group_idxs)
+    N = conf.shape[0]
+
+    params = nb.params
+
+    beta = nb.potential.beta
+    cutoff = nb.potential.cutoff
+    func = custom_ops.atom_by_atom_energies_f32
+    if precision == np.float64:
+        func = custom_ops.atom_by_atom_energies_f64
+
+    target_atoms = np.array(target_atoms).astype(np.int32)
+    comp_atoms = np.arange(N).astype(np.int32)
```

Comment:
rename `comp` -> `complete` or `all`

Reply:
(Does this test break if `comp_atoms` is shuffled?)

Reply:
This test does break if they are shuffled. We could pass the `all_atoms` into the wrapped function as well to make it work.

Reply:
okay, can we add a comment just saying this will break if shuffled?

Reply:
Renamed comp to all in https://github.com/proteneer/timemachine/pull/1204/commits/56c55c30294dd21e284d09f6cee01130f7f88e09

Added comment stating that shuffling breaks the test: https://github.com/proteneer/timemachine/pull/1204/commits/80a1945901496c6318afc49b452bf0402aa8be3a